xdem/volume.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+
+import xdem
+
+
+def hypsometric_binning(ddem: np.ndarray, ref_dem: np.ndarray, bin_size=50,
+                        normalized_bin_size: bool = False) -> pd.DataFrame:
+    """
+    Separate the dDEM in discrete elevation bins.
+
+    It is assumed that the dDEM is calculated as 'ref_dem - dem' (not 'dem - ref_dem').
+
+    :param ddem: The dDEM as a 2D or 1D array.
+    :param ref_dem: The reference DEM as a 2D or 1D array.
+    :param bin_size: The bin interval size in georeferenced units (or percent; 0-100, if normalized_bin_size=True)
+    :param normalized_bin_size: If the given bin size should be parsed as a percentage of the glacier's elevation range.
+
+    :returns: A Pandas DataFrame with elevation bins and dDEM statistics.
+    """
+    assert ddem.shape == ref_dem.shape
+    # Remove all nans, and flatten the inputs.
+    nan_mask = np.logical_or(
+        np.isnan(ddem) if not isinstance(ddem, np.ma.masked_array) else ddem.mask,
+        np.isnan(ref_dem) if not isinstance(ref_dem, np.ma.masked_array) else ref_dem.mask
+    )
+    ddem = ddem[~nan_mask]
```

Comment:
here we still need the values with NaN to quantify the area of each bin;
generally I do:
save the NaNs in the reference DEM -> save the count to a variable
keep the Nans in the dDEM (not NaN in reference DEM) -> save the count at each bin iteration (the interpolation per elevation bin happens here)
at the end, I propagate the mean volume change of the entire glacier to the reference DEM no data area

Reply:
You could save a 'total_count' for example.